File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    N = len(clothes)
    dict = {}
    
    for v,k in clothes:
        if dict.get(k):
            dict[k] += 1
            continue
        dict[k] = 1
    
    result = 1
    for v in dict.values():
        result *= (v+1)
    
    return result - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> map = new HashMap<>();
        
        for (String[] cloth : clothes) {
            String name = cloth[0];
            String type = cloth[1];
            map.put(type, map.getOrDefault(type, 0) + 1);
        }
        
        int result = 1;
        for (int v : map.values()){
            result *= (v+1);
        }
        
        return result - 1; // 아무것도 선택하지 않는 경우 1 차감
    }
}